### PR TITLE
[W01] Add Logging Tests

### PR DESCRIPTION
**Pull request by Flnn0**
#### General

- [x] I locally pass **all** tests or calculated manually the correct solutions on paper.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://github.com/JohannesStoehr/gad23-tests/blob/main/CONTRIBUTING.md).
- [x] The newly added tests do not reveal the solutions dynamically calculating the solutions as expected in the exercise.

### Description

- add `TestLogging.java`: checks whether the Logging result is right

#### Review
- [ ] Review 1
- [ ] Optional Review 2

### CODE REVIEW

**Comment by Anatoly03** [comment]
Passing all the tests! LGTM!

**Comment by Anatoly03** [comment]
I am passing all the tests! Great work! However it's bad practice to have code repetition, would you want to rewrite it to [parameterized](https://www.baeldung.com/parameterized-tests-junit-5) structure?

```java
@ParameterizedTest
@CsvSource({
     ............
})
public void test() {
    boolean[][] maze = Maze.generateMaze( ............ );
    int[] solutionX = new int[] { ............ };
    int[] solutionY = new int[] { ............ };
    ExampleResultImplementation result = new ExampleResultImplementation(solutionX.length);
    Walker walker = new Walker(maze, result);
    assertEquals(............, walker.walk(), "Walker should" + (............ ? " not " : " ") + "find exit");
    result.testLogging(solutionX, solutionY);
}
```

File: w01/test/gad/logging/TestLogging.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    @DisplayName("Logging - Artemis Example")
+    public void checkLoggingArtemisExample() {
+        boolean[][] maze = Maze.generateStandardMaze(10, 10);
+        int[] solutionX = new int[] { 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 3, 4, 4, 4, 3, 2, 1, 2, 3, 4, 4, 5, 6, 6, 7, 8, 9 };
+        int[] solutionY = new int[] { 0, 1, 2, 3, 4, 5, 6, 5, 5, 5, 6, 6, 7, 8, 8, 8, 8, 8, 8, 8, 7, 7, 7, 8, 8, 8, 8 };
+        ExampleResultImplementation result = new ExampleResultImplementation(solutionX.length);
+        Walker walker = new Walker(maze, result);
+        assertTrue(walker.walk(), "Walker should find exit");
+        result.testLogging(solutionX, solutionY);
+    }
```
Comment (Anatoly03) [inline w01/test/gad/logging/TestLogging.java:24]:
```suggestion
    @Test
    @DisplayName("Logging - Artemis Example")
    @MethodSource
    public void checkLogging(int width, int height, int seed, int[] solutionX, int[] solutionY, boolean expect) {
        boolean[][] maze = Maze.generateMaze(width, height, seed);
        
        ExampleResultImplementation result = new ExampleResultImplementation(solutionX.length);
        Walker walker = new Walker(maze, result);
        assertEquals(expect, walker.walk(), "Walker should" + (expect ? "" : " not") + " find exit");
        result.testLogging(solutionX, solutionY);
    }
    
    private static Stream<Arguments> checkLogging() {
        return Stream.of(
          Arguments.of(
              10, 10, 0
              new int[] { 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 3, 4, 4, 4, 3, 2, 1, 2, 3, 4, 4, 5, 6, 6, 7, 8, 9 },
              new int[] { 0, 1, 2, 3, 4, 5, 6, 5, 5, 5, 6, 6, 7, 8, 8, 8, 8, 8, 8, 8, 7, 7, 7, 8, 8, 8, 8 },
              true
          )
          //, More Arguments here
        );
    }
```

Comment (Flnn0) [reply w01/test/gad/logging/TestLogging.java]: I have changed my Tests to a parameterized structure.

**Review by meandaD — APPROVED** [review]
Hi! I ran the tests locally and I think they're good. The code structure is also great.

**Comment by Flnn0** [comment]
Thank you, have to change the README file and than it's finished from my side.

**Review by Anatoly03 — APPROVED** [review]

**Comment by Anatoly03** [comment]
LGTM!

**Comment by Anatoly03** [comment]
For the future something you can do is writing better commit messages, https://github.com/JohannesStoehr/gad23-tests/pull/16/commits/19176e7b8fcafc5e6f75ccb3e5f36b629f60dc53 could be “removed import” [what did you change in a commit] But that’s something you can only do in future pr’s :D

**Comment by Flnn0** [comment]
@Anatoly03 Thank you for your hint, it's my first time writing tests for git, thats no excuse. I will impruve in the future.

**Pull request merged**